### PR TITLE
Improve Test Coverage to 81%

### PR DESCRIPTION
**Pull request by yu-sun-77**
*Fixes #, if available:*

*Description of changes:*

1. Improve Test Coverage to 81%
2. exclude `ClusterSettingsManager.java` for coverage, not able to write UT for more coverage due to `java.net.ConnectException: Connection refused`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java [hunk 1]
```diff
@@ -30,7 +31,7 @@
     private boolean rcaEnabled;
     private boolean loggingEnabled;
     private boolean batchMetricsEnabled;
-    private volatile int shardsPerCollection;
+    @VisibleForTesting volatile int shardsPerCollection;
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java:34]: shardsPerCollection is not used by testing right ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/listener/PerformanceAnalyzerSearchListener.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public PerformanceAnalyzerSearchListener(final PerformanceAnalyzerController con
 
     @Override
     public String toString() {
-        return "PerformanceAnalyzerSearchListener";
+        return PerformanceAnalyzerSearchListener.class.getName();
```
Comment (rguo-aws) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/listener/PerformanceAnalyzerSearchListener.java:46]: should we use .getSimpleName() here instead ?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java (hunk 1 above)
Comment (yu-sun-77) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/PerformanceAnalyzerController.java]: good catch, reverting back

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterServiceEventMetricsTests.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    List<String> jsonStrs = readMetricsInJsonString();
+    assertTrue(jsonStrs.get(0).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_PRIORITY.toString()));
+    assertTrue(jsonStrs.get(1).contains(AllMetrics.MasterMetricValues.START_TIME.toString()));
+    assertTrue(jsonStrs.get(2).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_TYPE.toString()));
+    assertTrue(jsonStrs.get(3).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_METADATA.toString()));
+    assertTrue(jsonStrs.get(4).contains(AllMetrics.MasterMetricDimensions.MASTER_TASK_QUEUE_TIME.toString()));
+  }
+
+
+  private List<String> readMetricsInJsonString() {
```
Comment (aditjind) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterServiceEventMetricsTests.java:139]: Should we move this method to a util file and reuse it across the tests?

Comment (yu-sun-77) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterServiceEventMetricsTests.java]: Sure, moved to `TestUtil` file

**Review by aditjind — APPROVED** [review]

**Review by rguo-aws — APPROVED** [review]

**Pull request merged**